accounts/ethstore/src/account/safe_account.rs

```diff
@@ -77,16 +77,32 @@ impl SafeAccount {
 	/// Create a new `SafeAccount` from the given `json`; if it was read from a
 	/// file, the `filename` should be `Some` name. If it is as yet anonymous, then it
 	/// can be left `None`.
-	pub fn from_file(json: json::KeyFile, filename: Option<String>) -> Self {
-		SafeAccount {
+	pub fn from_file(json: json::KeyFile, filename: Option<String>) -> Result<Self, Error> {
+		SafeAccount::from_file_with_password(json, filename, &None)
+	}
+
+	/// Same as [`from_file`](#method.from_file), but with `password` which enables reading keystore files w/o address.
+	pub fn from_file_with_password(json: json::KeyFile, filename: Option<String>, password: &Option<Password>) -> Result<Self, Error> {
```

Comment:
I would make `password` mandatory here (see above comment). The decision which function to use and what errors to handle can be left to the caller.

Reply:
sure, it can be done, but for price of code repetition, because I use it to implement `from_file`